utils/s2n_mem.c

```diff
@@ -56,7 +56,7 @@ static int s2n_mem_init_impl(void)
 
     page_size = (uint32_t) sysconf_rc;
 
-    if (getenv("S2N_DONT_MLOCK")) {
+    if (getenv("S2N_DONT_MLOCK") || s2n_in_unit_test()) {
```

Comment:
For unit tests in our CMake file we already set `S2N_DONT_MLOCK=1`,  why do we need to check for `s2n_in_unit_test()` ? 

https://github.com/aws/s2n-tls/blob/main/CMakeLists.txt#L312

Reply:
It looks like the LGTM automated CI is ignoring the `S2N_DONT_MLOCK=1` environment variable in this PR, and is failing to scan the C code because of it: https://github.com/aws/s2n-tls/pull/2793#issuecomment-840170152

This change should work around this issue in LGTM code scanning.

Reply:
From the description of the issue:

> However, manually running unit tests without the test scripts (such as during debugging sessions, or by automated build tools not controlled by s2n) can cause the unit tests to fail due to most Linux OS's enforcing a low default limit to the amount of memory that can be locked at once.